src/Tizen.NUI.CommonUI/Controls/Button.cs

```diff
@@ -0,0 +1,1503 @@
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void OnFocusLost()
+        {
+            base.OnFocusLost();
+            UpdateState();
+        }
+        /// <summary>
+        /// Tap gesture event callback.
+        /// </summary>
+        /// <param name="source">Source which recieved touch event.</param>
+        /// <param name="e">Tap gesture event argument.</param>
+        /// <since_tizen> 6 </since_tizen>
+        /// This will be public opened in tizen_5.5 after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void OnTapGestureDetected(object source, TapGestureDetector.DetectedEventArgs e)
+        {
+            if (isEnabled)
+            {
+                ClickEventArgs eventArgs = new ClickEventArgs();
+                OnClick(eventArgs);
+                base.OnTapGestureDetected(source, e);
+            }
+        }
+
+        /// <summary>
+        /// Called after a touch event is received by the owning view.<br />
+        /// CustomViewBehaviour.REQUIRES_TOUCH_EVENTS must be enabled during construction. See CustomView(ViewWrapperImpl.CustomViewBehaviour behaviour).<br />
+        /// </summary>
+        /// <param name="touch">The touch event.</param>
+        /// <returns>True if the event should be consumed.</returns>
+        /// <since_tizen> 6 </since_tizen>
+        /// This will be public opened in tizen_5.5 after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool OnTouch(Touch touch)
+        {
+            PointStateType state = touch.GetState(0);
+      
+            switch(state)
+            {
+                case PointStateType.Down:
+                    isPressed = true;
+                    UpdateState();
+                    return true;
+                case PointStateType.Interrupted:
+                    isPressed = false;
+                    UpdateState();
+                    return true;
+                case PointStateType.Up:
+                    isPressed = false;
+                    if (buttonAttributes.IsSelectable != null && buttonAttributes.IsSelectable == true)
+                    {
+                        isSelected = !isSelected;
+                    }
+                    UpdateState();
+                    return true;
+                default:
+                    break;
+            }
+            return base.OnTouch(touch);
+        }
+        /// <summary>
+        /// Get Button attribues.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// This will be public opened in tizen_5.5 after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override Attributes GetAttributes()
+        {
+            return new ButtonAttributes();
```

Comment:
Is it correct to create a new Attribute?

Reply:
I think **GetAttributes**() is called when Control initialize its **attributes**. 
So returning new XXXAttriubutes seems to be correct.

Reply:
yes, it is correct.